### PR TITLE
feat: add support for Codeberg

### LINKED ISSUE #94: Add support for Codeberg
Hi! 👋 

Similarly to other providers like GitLab and sourcehut, I would like to propose _officially_ supporting [Codeberg](https://codeberg.org/). 

At the moment, as far as I've tested, it is possible to download a Codeberg repository (e.g., `npx tiged https://codeberg.org/joaopalmeiro/template-figma-plugin-script`), but support for Codeberg is not documented and there is no _shortcut_ for it: https://github.com/tiged/tiged/blob/c0537f7e4d9bec9d964be66b0b454c71f89ee572/src/index.ts#L772-L778

If I'm correct, it's even possible to download a Codeberg repository in `tar` mode since the following URL corresponds to the Codeberg one (e.g., https://codeberg.org/joaopalmeiro/template-figma-plugin-script/archive/main.tar.gz): https://github.com/tiged/tiged/blob/c0537f7e4d9bec9d964be66b0b454c71f89ee572/src/index.ts#L655

As an alternative, it is also possible to use the [Codeberg API](https://codeberg.org/api/swagger#/repository/repoGetArchive).

Let me know what you think, please. If you think supporting Codeberg makes sense, let me know if I can open a PR.

Thanks!

### PR DESCRIPTION
**Pull request by joaopalmeiro**
Closes #94

### CODE REVIEW

File: test/index.test.ts [hunk 1]
```diff
@@ -131,6 +131,17 @@ describe(degit, { timeout }, () => {
 		});
 	});
 
+	describe.sequential('Codeberg', () => {
+		it.each([
```
Comment (joaopalmeiro) [inline test/index.test.ts:135]: Hi, @nake89!

I think we need a Codeberg repo for testing. Are you going to create one on your side or do you want me to create one on my side?

Comment (nake89) [reply test/index.test.ts:135]: You can create one.

I might clone your repo to my codeberg later and change the test to match my repo url. But for now it is fine to just use yours.

Comment (joaopalmeiro) [reply test/index.test.ts:135]: Got it, thanks!

**Comment by nake89** [comment]
Thanks! This will be merged to version to 3.0.0 (not RC). I am awaiting aryaemami59 to update the changelog. I forgot to ask him to do it before the merge of the typescript PR. But I asked him now.  https://github.com/tiged/tiged/pull/89

After that is done. You can refresh your main branch and add your changes to CHANGELOG.md under version 3.0.0.

**Pull request merged**